app/src/main/java/com/keylesspalace/tusky/viewdata/StatusViewData.java

```diff
@@ -288,6 +300,32 @@ public boolean deepEquals(StatusViewData o) {
                     Objects.equals(card, concrete.card)
                     && isCollapsed == concrete.isCollapsed;
         }
+
+        static Spanned ReplaceCrashingCharacters(Spanned content) {
+            return (Spanned) ReplaceCrashingCharacters((CharSequence) content);
+        }
+
+        static CharSequence ReplaceCrashingCharacters(CharSequence content) {
+            Boolean replacing = false;
+            SpannableStringBuilder builder = null;
+
+            for (int index = 0; index < content.length(); ++index) {
```

Comment:
its not guaranteed that `length()` is a simple getter, moving it out of the loop should be faster